### PR TITLE
Make JupyterLab more self contained

### PR DESCRIPTION
**Pull request by blink1073**
Moves the ES6 promise polyfill, and font-awesome from the template to the webpack bundle.

We are still using MathJax and requirejs from bower.  It makes sense to only have one copy of MathJax in the repository since it is so big, and we'd might as well keep the known path to requirejs.

### CODE REVIEW

**Comment by blink1073** [comment]
cc @SylvainCorlay

File: notebook/lab/package.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
   },
   "devDependencies": {
     "css-loader": "^0.23.1",
-    "es6-promise": "^3.0.2",
+    "es6-promise": "^3.1.2",
```
Comment (sccolbert) [inline notebook/lab/package.json]: should this be a runtime dep?

Comment (SylvainCorlay) [inline notebook/lab/package.json]: So we discussed this on gitter and my reasoning for `jupyter-js-widgets` was that only the file that was intended to be webpacked explicitly required `es6-promise`, and webpacking is a build-time thing.

Comment (blink1073) [inline notebook/lab/package.json]: But it doesn't *need* to be Webpacked.

Comment (blink1073) [inline notebook/lab/package.json]: In general

Comment (SylvainCorlay) [inline notebook/lab/package.json]: Well, that is the thing (in jupyter-js-widgets), we have this `embed-webpack.js` which is not required by `index.js` and only intended to be webpacked.

**Comment by blink1073** [comment]
I'm going to mark this WIP, since currently the `lab.html` template already provides a polyfill from bower among other things.  I'm going to clean this up to be self contained.

**Comment by sccolbert** [comment]
So, I shouldn't hit the green button yet?

**Comment by blink1073** [comment]
Updated, see original comment.

File: notebook/templates/lab.html [hunk 2]
```diff
@@ -6,8 +6,6 @@
 
     <title>{{page_title}}</title>
     <meta http-equiv="X-UA-Compatible" content="IE=edge" />
-    <link rel="stylesheet" type="text/css" href="{{static_url("components/font-awesome/css/font-awesome.css")}}"></link>
```
Comment (SylvainCorlay) [inline notebook/templates/lab.html]: FYI, for requirejs, instead of using bower, you can load it with scriptjs.

Comment (blink1073) [inline notebook/templates/lab.html]: No, because the whole thing needs to be self contained.  (I used to work in a lab with no internet access).

**Comment by blink1073** [comment]
@SylvainCorlay, I finally got requirejs to play nicely with WebPack here:  88f9332.

File: notebook/templates/lab.html (hunk 2 above)
Comment (SylvainCorlay) [inline notebook/templates/lab.html]: Yes, you can load it locally with scripjs.

Comment (blink1073) [inline notebook/templates/lab.html]: Ah, I see.

File: notebook/templates/lab.html [hunk 3]
```diff
@@ -94,7 +92,6 @@
   "wsUrl": "{{ws_url| urlencode}}",
   "notebookPath": "{{notebook_path | urlencode}}"
 }</script>
-<script src="lab/bundle.js" type="text/javascript" charset="utf-8"></script>
```
Comment (minrk) [inline notebook/templates/lab.html]: Is it okay and/or necessary for lab to load before the require config?

Comment (blink1073) [inline notebook/templates/lab.html]: Argh!  Good point.

**Comment by blink1073** [comment]
@minrk, updated.

**Comment by minrk** [comment]
Looks okay to me. I'll let @SylvainCorlay merge in case he sees anything else.

**Comment by SylvainCorlay** [comment]
👍

**Pull request merged**